WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchBatchLoggerService.cs

```diff
@@ -0,0 +1,207 @@
+    {
+        mergeLogQueue.Enqueue(logData);
+    }
+
+    /// <inheritdoc />
+    public override async Task StopAsync(CancellationToken cancellationToken)
+    {
+        // Override StopAsync to wait for the log queue to be completely flushed before shutdown.
+        logger.LogInformation("BatchLogger is stopping. Flushing remaining logs.");
+        await FlushUntilEmptyAsync();
+        await base.StopAsync(cancellationToken);
+    }
+
+    /// <inheritdoc />
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        // Continuously flush log entries until cancellation is requested.
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            await FlushAsync();
+            try
+            {
+                await Task.Delay(options.FlushInterval, stoppingToken);
+            }
+            catch (TaskCanceledException)
+            {
+                // Ignore cancellation exceptions from Task.Delay.
+            }
+        }
+
+        // When cancellation is requested, flush any remaining entries.
+        await FlushUntilEmptyAsync();
+    }
+
+    /// <summary>
+    /// Method to flush the log queue until it is empty.
+    /// This should be called when the application is stopping to ensure all logs are written before the application is closed.
+    /// </summary>
+    private async Task FlushUntilEmptyAsync()
+    {
+        // Loop until the queue is empty or no new logs are added.
+        while (!mergeLogQueue.IsEmpty)
+        {
+            await FlushAsync();
+        }
+    }
+
+    /// <summary>
+    /// Flush the next batch of log entries to the database.
+    /// </summary>
+    private async Task FlushAsync()
+    {
+        // Create a list for the next batch of logs.
+        var entriesToInsert = new List<BranchMergeLogModel>();
+        string branchDatabaseConnectionString = null;
+
+        // Try to dequeue log entries from the queue until we reach the batch size, or the connection string changes, or the queue is empty.
+        while (entriesToInsert.Count < options.BatchSize
+               && (
+                   branchDatabaseConnectionString == null
+                   || (mergeLogQueue.TryPeek(out var logEntry) && branchDatabaseConnectionString == logEntry.BranchDatabaseConnectionString)
+               )
+               && mergeLogQueue.TryDequeue(out logEntry))
+        {
+            branchDatabaseConnectionString = logEntry.BranchDatabaseConnectionString;
+            entriesToInsert.Add(logEntry);
+        }
+
+        // If there are no entries to insert, return early.
+        if (entriesToInsert.Count == 0)
+        {
+            return;
+        }
+
+        try
+        {
+            // Use the connection string from the first entry in the batch to create a MySQL connection.
+            await using var mySqlConnection = new MySqlConnection(entriesToInsert.First().BranchDatabaseConnectionString);
```

Comment:
A MySQL connection is created hard coded, is it possible to use the IDatabaseConnection in combination with an IServiceProvider to create a temporary scope? The interfaces provides and option to change the connection string.

Reply:
You are right, that didn't occur to me for some reason. I changed it now (and tested).